paasta_tools/kubernetes/remote_run.py

```diff
@@ -0,0 +1,373 @@
+    :param str service_account: service account name
+    :return: token value
+    """
+    token_spec = V1TokenRequestSpec(
+        expiration_seconds=600,  # minimum allowed by k8s
+        audiences=[],
+    )
+    request = AuthenticationV1TokenRequest(spec=token_spec)
+    response = kube_client.core.create_namespaced_service_account_token(
+        service_account, namespace, request
+    )
+    return response.status.token
+
+
+def create_remote_run_service_account(
+    kube_client: KubeClient,
+    namespace: str,
+    pod_name: str,
+    user: str,
+) -> str:
+    """Create service account to exec into remote-run pod
+
+    :param KubeClient kube_client: Kubernetes client
+    :param str namespace: pod namespace
+    :param str pod_name: pod name
+    :param str user: user requiring credentials
+    """
+    pod_name_hash = hashlib.sha1(pod_name.encode("utf-8")).hexdigest()[:12]
+    service_account_name = f"remote-run-{user}-{pod_name_hash}"
+    service_accounts = get_all_service_accounts(kube_client, namespace=namespace)
+    if any(item.metadata.name == service_account_name for item in service_accounts):
+        return service_account_name
+    service_account = V1ServiceAccount(
+        metadata=V1ObjectMeta(
+            name=service_account_name,
+            namespace=namespace,
+            labels={"paasta.yelp.com/pod_owner": user},
+        )
+    )
+    kube_client.core.create_namespaced_service_account(
+        namespace=namespace, body=service_account
+    )
+    return service_account_name
+
+
+def create_pod_scoped_role(
+    kube_client: KubeClient,
+    namespace: str,
+    pod_name: str,
+    user: str,
+) -> str:
+    """Create role with execution access to specific pod
+
+    :param KubeClient kube_client: Kubernetes client
+    :param str namespace: pod namespace
+    :param str pod_name: pod name
+    :param str user: user requiring the role
+    :return: name of the role
+    """
+    pod_name_hash = hashlib.sha1(pod_name.encode("utf-8")).hexdigest()[:12]
+    role_name = f"remote-run-role-{pod_name_hash}"
+    policy = V1PolicyRule(
+        verbs=["create", "get"],
+        resources=["pods", "pods/exec"],
+        resource_names=[pod_name],
+        api_groups=[""],
+    )
+    role = V1Role(
+        rules=[policy],
+        metadata=V1ObjectMeta(
+            name=role_name,
+            labels={"paasta.yelp.com/pod_owner": user},
+        ),
+    )
+    kube_client.core.create_namespaced_role(namespace=namespace, body=role)
+    return role_name
+
+
+def bind_role_to_service_account(
+    kube_client: KubeClient,
+    namespace: str,
+    service_account: str,
+    role: str,
+) -> None:
+    """Bind service account to role
+
+    :param KubeClient kube_client: Kubernetes client
+    :param str namespace: service account namespace
+    :param str service_account: service account name
+    :param str role: role name
+    """
+    role_binding = V1RoleBinding(
```

Comment:
should we add some labels to this rolebinding so that we can more easily clean up later?